ticket/DM-33736

b1quint opened the pull request.

b1quint committed:
Add Sky Almanac.ipynb

patrickingraham commented on procedures/observing_tools/Sky Almanac.ipynb:
```diff
@@ -0,0 +1,697 @@
+{
```
<p>It does more than that :)</p><p>It also plots the elevation and airmass for a series of designated targets!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

patrickingraham replied:
<p>State your source here... ideally it'd come out of an observatory-wide constant file (see https://github.com/lsst-ts/ts_observatory_control/tree/develop/python/lsst/ts/observatory/control/constants) but I'm not sure we have that yet.</p><p>https://github.com/lsst-ts/ts_pointing_common/blob/develop/src/data/MTPtg.info#L25 works.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

patrickingraham replied:
<p>You can remove the commented line.</p><p>Maybe add a line above this that says something like, "Declare targets for which the sky coordinates will be calculated"</p><p> </p><p>Also, "Beginning of the night"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

patrickingraham replied:
<pre class="ql-syntax" spellcheck="false">Line #27.        Get the airmass for a given target at a given location and at a given time.
</pre><p>Mention which model you're using to do this (plane parallel atmosphere)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

patrickingraham replied:
<p>Consider, "Plotting Target Airmass vs Time"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

patrickingraham replied:
<p>Consider, "Plotting Target Elevation vs Time"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

patrickingraham replied:
<p>Why not just have this plot, but then put the airmass on the axis on the right?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='19'/>

patrickingraham commented:
This is really cool.
As a next step, it would be really cool to calculate the angle between the moon and the target. Essentially the elevation sometimes looks like it'll be the same as the moon, except the moon could be on the other side of the sky...
I would imagine Astropy has a fxn to do that math... or you can try to remember how to do spherical trig :)

tribeiro commented:
One thing to consider is moving the code in the notebook to a package like [ts_observing_utilities](https://github.com/lsst-ts/ts_observing_utilities)... you may want to take a look at [astroplan](https://astroplan.readthedocs.io/en/latest/tutorials/plots.html) though, it has a lot of useful tools...

b1quint committed:
Update Sky Almanac.ipynb

b1quint commented:
I had a look at astroplan. However, my Nublado does not have it installed by default and I tried to create a notebook with the default packages. Maybe we should include astroplan to the default packages but I did not want to rely on that now. 

I pushed a new commit addressing all the comments from @patrickingraham. I cleaned up the outputs too.

patrickingraham reviewed (approved):
Thanks for making the changes. Looks great!
